python_modules/dagster/dagster/_core/execution/calculate_data_time.py

```diff
@@ -0,0 +1,125 @@
+from datetime import datetime, timezone
+from typing import AbstractSet, Mapping, Optional, Tuple
+
+from ..definitions.events import AssetKey
+from ..instance import DagsterInstance
+
+ROOT_DATA_TAG = ".dagster/root_data_ids"
```

Comment:
"root" and "data" aren't terms that are still used in the remainder of the freshness policy literature, right? Is there a name that would be more in-keeping?

Also, haven't thought this through at all, but did you consider having a tag per root asset? It could make it easier to find the downstream materializations associated with a materialization, if that's something we want to do in the future.